internal/services/fluidrelay/validate/server_name.go

```diff
@@ -0,0 +1,21 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var serverNameReg = regexp.MustCompile(`^[-0-9a-zA-Z]{1,50}$`)
```

Comment:
because hyphens are acceptable but underscore is not allowed
<img width="393" alt="image" src="https://user-images.githubusercontent.com/2633022/173543606-8345d340-d162-48dc-8ac2-1eab2d65be4c.png">